force-app/main/default/classes/TaskTriggerHandler.cls

```diff
@@ -0,0 +1,76 @@
+public without sharing class TaskTriggerHandler {
+  public static void updateRelatedAccounts(List<Task> tasks) {
+    Set<Id> whatIdsRelatedToAccounts = new Set<Id>();
+    for (Task task : tasks) {
+      if (
+        task.WhatId <> null &&
+        task.WhatId.getSobjectType() == Account.getSObjectType()
```

Comment:
nice one;
there's also 

```
Schema.Account.SObjectType.getDescribe()
      .getKeyPrefix();
```

Your's is better. 

You can use the safe navigation operator instead of the null check

```
task.WhatId?.getSobjectType()
```